storage/remote/queue_manager.go

```diff
@@ -1084,15 +1097,32 @@ func sendWriteRequestWithBackoff(ctx context.Context, cfg config.QueueConfig, l
 
 		backoff = sleepDuration * 2
 
-		if backoff > cfg.MaxBackoff {
-			backoff = cfg.MaxBackoff
+		if backoff > cfg.Retry.MaxBackoff {
+			backoff = cfg.Retry.MaxBackoff
 		}
 
 		try++
 		continue
 	}
 }
 
+func shouldAttempt(policy config.RetryPolicy, minTs int64, retries int) bool {
+	if !policy.IsSet() {
```

Comment:
`IsSet` still doesn't really make sense to me. Could the RetryPolicy field on the config struct be a pointer, and we just check if the pointer is nil?